### PR TITLE
Ensure manually created OpenTelemetry spans are compliant with trace metrics

### PR DESCRIPTION
**Pull request by mcculls**
# What Does This Do

Translates the `http.response.status_code` attribute (OpenTelemetry) to `http.status_code` (Datadog)

# Additional Notes

This PR also fixes an issue where the `AttributeKey` form of the `setAttribute` method was not covered for non-array reserved attributes.

### CODE REVIEW

**Review by amarziali — APPROVED** [review]

**Review by zarirhamza — CHANGES_REQUESTED** [review]
`http.request.status_code` is an int value as per [otel](https://opentelemetry.io/docs/specs/semconv/attributes-registry/http/) 

BUT `http.status_code` is a string value following datadog standards

Just want to confirm, does this PR follow those standards?

**Comment by mcculls** [comment]
Hi @zarirhamza - yes, `setHttpStatusCode` is the method used by all the web integrations.

If you want to follow the sequence it goes:
* [DDSpan.setHttpStatusCode](https://github.com/DataDog/dd-trace-java/blob/v1.34.0/dd-trace-core/src/main/java/datadog/trace/core/DDSpan.java#L508)
* [DDSpanContext.setHttpStatusCode](https://github.com/DataDog/dd-trace-java/blob/v1.34.0/dd-trace-core/src/main/java/datadog/trace/core/DDSpanContext.java#L670)
* that special `httpStatusCode` is then treated specially when it's sent as metadata
* [DDSpanContext.processTagsAndBaggage](https://github.com/DataDog/dd-trace-java/blob/v1.34.0/dd-trace-core/src/main/java/datadog/trace/core/DDSpanContext.java#L811)
* [TraceMapperV0_4.accept](https://github.com/DataDog/dd-trace-java/blob/bff441bbdf82a488e4690abc96352fc0b283621f/dd-trace-core/src/main/java/datadog/trace/common/writer/ddagent/TraceMapperV0_4.java#L121)

TL;DR is that this is the same method used by Datadog integrations in the Java tracer to set the HTTP statuis for a span, so given that is producing the right payloads this will also produce the expected payload.

**Review by zarirhamza — APPROVED** [review]

**Review by PerfectSlayer — COMMENTED** [review]
> This PR also fixes an issue where the AttributeKey form of the setAttribute method was not covered for non-array reserved attributes.

Nice catch. Do you think I should also improve the original convention tests to check builder and attribute keys too?

**Comment by mcculls** [comment]
> Nice catch. Do you think I should also improve the original convention tests to check builder and attribute keys too?

Yes, it would be good to do at some point

**Comment by PerfectSlayer** [comment]
I just made a PR about it.
